clustercontroller-core/src/test/java/com/yahoo/vespa/clustercontroller/core/DatabaseHandlerTest.java

```diff
@@ -104,4 +133,29 @@ public void empty_bundle_is_returned_if_no_db_connection() throws Exception {
         assertEquals(ClusterStateBundle.empty(), retrievedBundle);
     }
 
+    @Test
+    public void save_wanted_state_sheds_orphaned_states() throws Exception {
+        var fixture = new Fixture();
+        DatabaseHandler handler = fixture.createHandler();
+        DatabaseHandler.DatabaseContext databaseContext = fixture.createMockContext();
+
+        // The test fixture contains 10 nodes with indices 1-10.  A wanted state for
+        // an existing node (5) should be preserved.  Note that it is not possible to set a
+        // wanted state outside the existing nodes.
+        Node storageNode5 = Node.ofStorage(5);
+        NodeState maintenance = new NodeState(NodeType.STORAGE, State.MAINTENANCE);
+        databaseContext.getCluster().getNodeInfo(storageNode5).setWantedState(maintenance);
+        var expectedWantedStates = new TreeMap<>(Map.of(storageNode5, maintenance));
+
+        // Ensure database is connected to ZooKeeper
+        assertTrue(handler.doNextZooKeeperTask(databaseContext));
+
+        // Verify saveWantedStates() is invoked once
```

Comment:
Should this be "Verify storeWantedStates() is invoked once", not `saveWantedStates`?

Reply:
Done